Xamarin.Forms.Core/ToolbarItem.cs

```diff
@@ -12,11 +12,34 @@ public class ToolbarItem : MenuItem
 
 		static readonly BindableProperty PriorityProperty = BindableProperty.Create("Priority", typeof(int), typeof(ToolbarItem), 0);
 
+		public static readonly BindableProperty IsVisibleProperty = BindableProperty.Create("IsVisible", typeof(bool), typeof(ToolbarItem), true, BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) =>
+		{
+			var item = bindable as ToolbarItem;
+
+			if (item != null && item.Parent == null)
+				return;
+
+			if (item != null)
+			{
+				var items = ((ContentPage)item.Parent).ToolbarItems;
+
+				if ((bool)newvalue && !items.Contains(item))
+				{
+					items.Add(item);
```

Comment:
Any reason why this would not be ```Device.BeginInvokeOnMainThread(() => { items.Add(item); });``` like it is [here](https://forums.xamarin.com/discussion/comment/306901/#Comment_306901) to guarantee it will run on the UI thread.